app/authentication/server/startup/index.js

```diff
@@ -278,26 +276,18 @@ Accounts.insertUserDoc = _.wrap(Accounts.insertUserDoc, function (insertUserDoc,
 		}
 	}
 
-	if (noRoles || roles.length === 0) {
-		const hasAdmin = Users.findOne(
-			{
-				roles: 'admin',
-				type: 'user',
-			},
-			{
-				fields: {
-					_id: 1,
-				},
-			},
-		);
-
-		if (hasAdmin) {
-			roles.push('user');
-		} else {
-			roles.push('admin');
-			if (settings.get('Show_Setup_Wizard') === 'pending') {
-				Settings.updateValueById('Show_Setup_Wizard', 'in_progress');
-			}
+	/**
+	 * if settings shows setup wizard to be pending
+	 * and no admin's been found,
+	 * and existing role list doesn't include admin
+	 * create this user admin.
+	 * count this as the completion of setup wizard step 1.
+	 */
+	const hasAdmin = Users.findOneByRolesAndType('admin', 'user', { fields: { _id: 1 } });
+	if (!roles.includes('admin') && !hasAdmin) {
+		roles.push('admin');
+		if (settings.get('Show_Setup_Wizard') === 'pending') {
+			Settings.updateValueById('Show_Setup_Wizard', 'in_progress');
 		}
 	}
```

Comment:
```suggestion
	/**
	 * if settings shows setup wizard to be pending
	 * and existing role list doesn't include admin
	 * create this user admin.
	 * count this as the completion of setup wizard step 1.
	 */
	if (settings.get('Show_Setup_Wizard') === 'pending') {
		if (roles.includes('admin')) {
		    roles.push('admin');
		}
		Settings.updateValueById('Show_Setup_Wizard', 'in_progress');
	}
```

Hey @matheusbsilva137 - take a look at this, https://github.com/RocketChat/Rocket.Chat/blob/443866e706602a2cf33a56e4e3b25ccacf7e7c42/server/startup/initialData.js#L138-L145

No admin situation is already handled at startup, so I don't think we need to do re-do that here. Could be wrong. I'm just leaving this  as a suggestion here.

Reply:
I've just realized this approach could create a security issue 🤔 
Since the `Show_Setup_Wizard` setting is _not hidden_, a user may have the permission to edit it (eg to `pending`) and, then, create a new user who will become an admin (even though the server already has an admin user).